### ISSUE TITLE
StrictMode Complaining

### ISSUE THREAD

**Issue by timrijckaert** [Adobe-Marketing-Cloud/mobile-services#258]
Hi, 

We are using the `StrictMode` feature of Android and saw some logs passing by from Adobe Analytics you might want to fix.

```
StrictMode policy violation; ~duration=21 ms: android.os.StrictMode$StrictModeDiskReadViolation: policy=1114175 violation=2
at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1440)
at java.io.UnixFileSystem.checkAccess(UnixFileSystem.java:251)
at java.io.File.exists(File.java:807)
at android.app.ContextImpl.ensurePrivateDirExists(ContextImpl.java:572)
at android.app.ContextImpl.ensurePrivateCacheDirExists(ContextImpl.java:568)
at android.app.ContextImpl.getCacheDir(ContextImpl.java:661)
at android.content.ContextWrapper.getCacheDir(ContextWrapper.java:262)
at com.adobe.mobile.StaticMethods.getCacheDirectory(StaticMethods.java:599)
at com.adobe.mobile.AnalyticsWorker.<init>(AnalyticsWorker.java:77)
at com.adobe.mobile.AnalyticsWorker.sharedInstance(AnalyticsWorker.java:64)
at com.adobe.mobile.MobileConfig$13.onReceive(MobileConfig.java:1244)
at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$-android_app_LoadedApk$ReceiverDispatcher$Args_52497(LoadedApk.java:1313)
at android.app.-$Lambda$aS31cHIhRx41653CMnd4gZqshIQ.$m$7(Unknown Source:4)
at android.app.-$Lambda$aS31cHIhRx41653CMnd4gZqshIQ.run(Unknown Source:39)
at android.os.Handler.handleCallback(Handler.java:790)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6494)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by sbenedicadb** [comment]
Hey @timrijckaert,

Would you mind letting me know which version you see this on so I can create an internal ticket for us to work on this?  Thanks.

**steve benedick**
*sr mobile software engineer*
[Adobe Experience Cloud](http://www.adobe.com/experience-cloud.html)

**Comment by timrijckaert** [comment]
4.14.0

**Comment by shalehaha** [comment]
@timrijckaert,

SDK is relying on the access to the file system to properly working, it stores cached files and databases files on the disk. So during you test with StrictMode, you should enable disk read and disk write for it.  
Hope this helps, let us know if you have any concern. Thanks.


**Jiabin Geng**
*mobile software engineer*
[Adobe Marketing Cloud](http://www.adobe.com/solutions/digital-marketing.html)

**Comment by timrijckaert** [comment]
I agree however this should not happen on the main thread hence Strictmode complaining about it.

**Comment by shalehaha** [comment]
@timrijckaert,

Thanks, I see the problem. We will be working on a fix and will update here once we have a release date scheduled for it.

**Jiabin Geng**
*mobile software engineer*
[Adobe Marketing Cloud](http://www.adobe.com/solutions/digital-marketing.html)

**Comment by shalehaha** [comment]
@timrijckaert,

This issue should be fixed with the 4.15.0 release, could you please verify it. Thanks.
